### PR TITLE
ED-6475 - Updates to support selecting rows with checkboxes

### PR DESCRIPTION
**Pull request by cheeverm**

### CODE REVIEW

File: src/table/index.js [hunk 1]
```diff
@@ -57,36 +57,10 @@ function getFieldValue(cellValue, type, format, dataFormat) {
   }
   return cellValue;
 }
-function isEquivalentObject(a, b) {
```
Comment (sneu012) [inline src/table/index.js:60]: Is this no longer used?

File: src/table/index.js [hunk 2]
```diff
@@ -158,29 +133,32 @@ class TableField extends Component {
...
   }
   handleRowSelect(row, isSelected, e) {
+    const { onChange } = this.props;
+
     const {
-      data,
+      data = [],
       selectRow: { onSelectRow: { fieldToUpdate = "picked" } }
     } = this.tableConf;
-    let filteredRows = (data || []).map(item => {
```
Comment (sneu012) [inline src/table/index.js:165]: Just wanted to double check that this wouldn't break any existing feature

Comment (cheeverm) [reply src/table/index.js:165]: Confirmed that the built-in select all checkbox still works using the medications form

File: src/table/index.js (hunk 1 above)
Comment (cheeverm) [reply src/table/index.js:60]: Reworked the one function that was using this in the form extras library

**Review by sneu012 — APPROVED** [review]
LGTM!

File: playground/app/procedureCodes/index.js [hunk 3]
```diff
@@ -6,73 +6,12 @@ export default Object.assign({}, schema, uiSchema, rules, {
   formData: {
     procedureCodes: [
       {
-        diagnosis: {
```
Comment (yusuff-rxnt) [inline playground/app/procedureCodes/index.js:9]: please keep this json to load precedurecode plugin with features.

File: src/table/index.js [hunk 4]
```diff
@@ -158,29 +133,32 @@ class TableField extends Component {
...
   handleAllRowSelect(isSelected, rows, e) {
     const {
-      // data,
+      data,
       selectRow: { onSelectAllRow: { fieldToUpdate = "picked" } }
     } = this.tableConf;
 
-    let filteredRows = (rows || []).map(item => {
+    let filteredRows = data.map(item => {
+      item["isSelected"] = isSelected;
```
Comment (yusuff-rxnt) [inline src/table/index.js:161]: i think we have used isSelected index in the plugin sync.(patient dashboard sync).
 please make sure its not affect the existing functionality.

File: playground/app/procedureCodes/index.js (hunk 3 above)
Comment (cheeverm) [reply playground/app/procedureCodes/index.js:9]: Added back

File: src/table/index.js (hunk 4 above)
Comment (cheeverm) [reply src/table/index.js:161]: Switch property name

**Pull request merged**